doc/kernel/memory/pools.rst

```diff
@@ -168,6 +173,25 @@ memory block is actually used to satisfy the request.)
     ... /* use memory block */
     k_mem_pool_free(&block);
 
+Thread Resource Pools
+*********************
+
+Certain kernel APIs may need to make heap allocations on behalf of the
+calling thread. For example, some initialization APIs for objects like
+pipes and message queues may need to allocate a private kernel-side buffer,
+or objects like queues may temporarily allocate kernel data structures
+temporarily as items are placed in the queue.
```

Comment:
one too many uses of ``temporarily`` in this sentence -- I'd delete this last one.